File: RentIt.Mobile/src/domains/rentals/domain/rental.ts
import type { Equipment } from '@/src/shared/domain/equipment';
import type { RentalStatus } from './status';

export interface RentalClient {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
}

export interface Rental {
  id: number;
  dateFrom: string;
  dateTo: string;
  notes?: string;
  address: string;
  clientId: number;
  equipmentId: number;
  statusId: number;
  client?: RentalClient;
  equipment?: Equipment;
  status?: RentalStatus;
}

File: RentIt.Server/Models/Rental.cs
namespace RentIt.Server.Models;

public class Rental
{
    public int Id { get; set; }
    public DateTime DateFrom { get; set; }
    public DateTime DateTo { get; set; }
    public string Notes { get; set; } = string.Empty;

    public string Address { get; set; } = string.Empty;

    public int ClientId { get; set; }
    public int EquipmentId { get; set; }
    public RentalStatus Status { get; set; } = RentalStatus.Pending;

    public User? Client { get; set; }
    public Equipment? Equipment { get; set; }
}
